checkov/terraform_json/runner.py

```diff
@@ -0,0 +1,230 @@
+from checkov.terraform.checks.resource.registry import resource_registry
+from checkov.terraform.graph_builder.local_graph import TerraformLocalGraph
+from checkov.terraform.runner import Runner as TerraformRunner
+from checkov.terraform_json.utils import get_scannable_file_paths, TF_JSON_POSSIBLE_FILE_ENDINGS, create_definitions
+
+if TYPE_CHECKING:
+    from checkov.common.graph.checks_infra.registry import BaseRegistry
+    from checkov.terraform.graph_manager import TerraformGraphManager
+    from checkov.common.typing import LibraryGraphConnector, _CheckResult
+
+logger = logging.getLogger(__name__)
+
+
+class Runner(TerraformRunner):
+    check_type = CheckType.TERRAFORM_JSON  # noqa: CCE003  # a static attribute
+
+    def __init__(
+        self,
+        graph_class: type[TerraformLocalGraph] = TerraformLocalGraph,
+        graph_manager: TerraformGraphManager | None = None,
+        db_connector: LibraryGraphConnector | None = None,
+        external_registries: list[BaseRegistry] | None = None,
+        source: str = GraphSource.TERRAFORM,
+    ) -> None:
+        super().__init__(
+            graph_class=graph_class,
+            graph_manager=graph_manager,
+            db_connector=db_connector,
+            external_registries=external_registries,
+            source=source,
+        )
+        self.file_extensions = TF_JSON_POSSIBLE_FILE_ENDINGS  # override what gets set from the TF runner
+        self.graph_registry = get_graph_checks_registry(super().check_type)
+
+        self.definitions = {}
+        self.context = {}
+        self.root_folder: str | None = None
+
+    def run(
+        self,
+        root_folder: str | None = None,
+        external_checks_dir: list[str] | None = None,
+        files: list[str] | None = None,
+        runner_filter: RunnerFilter | None = None,
+        collect_skip_comments: bool = True,
+    ) -> Report | list[Report]:
+        runner_filter = runner_filter or RunnerFilter()
+
+        if not runner_filter.show_progress_bar:
+            self.pbar.turn_off_progress_bar()
+
+        report = Report(self.check_type)
+        self.root_folder = root_folder
+
+        if not self.definitions or not self.context:
+            file_paths = get_scannable_file_paths(
+                root_folder=root_folder, files=files, excluded_paths=runner_filter.excluded_paths
+            )
+
+            if not file_paths:
+                return report
+
+            self.definitions, self.definitions_raw, parsing_errors = create_definitions(file_paths)
+
+            report.add_parsing_errors(parsing_errors)
+
+            if external_checks_dir:
+                for directory in external_checks_dir:
+                    resource_registry.load_external_checks(directory)
+
+                    if CHECKOV_CREATE_GRAPH:
+                        self.graph_registry.load_external_checks(directory)
+
+            # TODO: create function 'build_definitions_context()'
```

Comment:
should this comments stay here? If yes, is there a linked issue for it (or do we want to create one)?

Reply:
#4635 4635